### ISSUE TITLE
FEAT [COMMENT'S BADGE]

### ISSUE THREAD

**Issue by beldier** [solex2006/SELIProject#211]
# INDEX
1 [REQUIREMENT](#requirement)

1.1 [USER STORIES](#user-stories)

1.2 [WORKFLOW](#workflow)

1.3 [REQUIREMENTS (DETAILS)](#requirement-details)

1.4 [WIREFRAMES](#wireframes)

2 [RESOURCES](#resources)

2.1 [TOOLS](#tools)

2.2 [SUGGESTED IMPLEMENTATION](#suggested-implementation)

2.3 [REFERENCES](#references)

3 [GLOSSARY](#glossary)

# REQUIREMENT

## USER STORIES
As a tutor is required to award a badge based on best comment in forum activiy 

## WORKFLOW
![flowComments](https://user-images.githubusercontent.com/36335128/92510788-5d60a200-f1da-11ea-941e-66677773a8d1.png)

## REQUIREMENT DETAILS
tutor - > Published courses - > Classroom management -> See course -> Actitivy(forum)
It is required to supply a way to award a badge based on the best comment of the activity . 

## WIREFRAMES
First we need to implement the 3-dots button in comments to display a menu with badge generation
![Frame 8](https://user-images.githubusercontent.com/36335128/92514495-0067ea80-f1e0-11ea-87e8-281ffd31bb28.png)

Second will be to show the component to create badge , currently is nested into quizForm.js code, so extract the code into a component and use it in forum activity
![Frame 9](https://user-images.githubusercontent.com/36335128/92514598-2a211180-f1e0-11ea-8dea-a92fad13b712.png)

Last will be to display some sort of insignia in the comment, to distinguish that it already has a badge 
![Frame 7](https://user-images.githubusercontent.com/36335128/92514750-6e141680-f1e0-11ea-9ef4-c82e110eba8f.png)

# RESOURCES

## SUGGESTED IMPLEMENTATION
The place to store the badge is critical, since it should work with data already in the platform 
The next document it is activities collection (db.activities.find().pretty())
```
{
	"_id" : "LjTok7ybL3QGcpcmz",
	"activity" : {
		"data" : [
			{
				"id" : 0.31611039798347473,
				"userId" : "nXjoZYCaKfqCmv32m",
				"date" : ISODate("2020-09-04T16:37:17.575Z"),
				"label" : "Hello 1st comment",
				"media" : [ ]
                              //"badge":"Badge Object".      Probably best place to store each comment badge, up to discussion

			},
			{
				"id" : 0.9680622785060196,
				"userId" : "nXjoZYCaKfqCmv32m",
				"date" : ISODate("2020-09-04T16:37:24.389Z"),
				"label" : "Hello 2nd comment",
				"media" : [ ]
                              //"badge":"Badge Object".      Probably best place to store each comment badge, up to discussion

			},
			{
				"id" : 0.8890996896046655,
				"userId" : "A5ZezmMc7s6KdyHpg",
				"date" : ISODate("2020-09-04T16:39:01.875Z"),
				"label" : "my first comment",
				"media" : [ ]
                              //"badge":"Badge Object".      Probably best place to store each comment badge, up to discussion

			},
			{
				"id" : 0.6891948494688911,
				"userId" : "nXjoZYCaKfqCmv32m",
				"date" : ISODate("2020-09-04T16:39:57.013Z"),
				"label" : "3rd comment",
				"media" : [ ]
                              //"badge":"Badge Object".      Probably best place to store each comment badge, up to discussion
			}
		],
		"type" : "forum",
		"public" : false,
		"date" : ISODate("2020-09-04T16:03:52.065Z"),
		"user" : "dcAnko74fnwbFfTLN",
		"course" : "q3Nvrda3ZMWxfk4Mg"
	}
}
```
## REFERENCES

- [Currently documentation of seli open badges ](https://docs.google.com/document/d/1D0rMAyY7eaNcFIY9cQHEWIMWaCihJgUQSkSGZN5RVqs/edit?usp=sharing)

*******

# GLOSSARY

**Comment by ecureuill** [comment]
Hello @beldier this feature is going to be developed in current release or this need to be discussed yet?

**Comment by beldier** [comment]
yes @ecureuill, sorry forgot that label , is currently in development
